src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CartRuleController.php

```diff
@@ -261,6 +285,55 @@ public function bulkDisableAction(Request $request): RedirectResponse
         return $this->redirectToRoute('admin_cart_rules_index');
     }
 
+    /**
+     * This just prefills form fields that depends on query parameter like customer search input (if id is provided),
+     * all remaining data should be set in related form data provider
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    private function prefillFormData(Request $request): array
```

Comment:
this pollutes the controller abit, but Im not sure if its worth moving this somewhere else. The format of customer is already used in couple places (basically where CustomerSearchInput is used), maybe its worth extracting some provider to do that, but im not sure, it might be too specific :man_shrugging: